x-pack/plugins/security_solution/public/app/home/index.tsx

```diff
@@ -66,3 +76,74 @@ const HomePageComponent: React.FC<HomePageProps> = ({
 HomePageComponent.displayName = 'HomePage';
 
 export const HomePage = React.memo(HomePageComponent);
+
+const useInitSearchBarUrlParams = () => {
+  const dispatch = useDispatch();
+  const { filterManager, savedQueries } = useKibana().services.data.query;
+
+  const onInitializeAppQueryUrlParam = useCallback(
+    (initialState: Query | null) => {
+      if (initialState != null) {
+        dispatch(
+          inputsActions.setFilterQuery({
+            id: 'global',
+            query: initialState.query,
+            language: initialState.language,
+          })
+        );
+      }
+    },
+    [dispatch]
+  );
+
+  const onInitializeFiltersUrlParam = useCallback(
+    (initialState: Filter[] | null) => {
+      if (initialState != null) {
+        dispatch(
+          inputsActions.setSearchBarFilter({
+            id: 'global',
+            filters: initialState,
+          })
+        );
+
+        filterManager.setFilters(initialState);
+      } else {
+        // Clear filters to ensure that other App filters don't leak into security solution.
+        filterManager.setFilters([]);
```

Comment:
The changes here look really clean, thanks for doing this! One thing, on investigations, we've noticed a lot of weird behavior around the filter manager in general that led to this issue being made: https://github.com/elastic/kibana/issues/124695 . You may want to look into some of the `filterManager` functionality that already exists to verify the expected functionality (i.e. allowing filters to be pinned across solutions). For instance this functionality no longer works on this branch. Thanks again for the awesome work!

https://user-images.githubusercontent.com/17211684/177340449-9229c448-f0a6-4c60-bafc-66212ef59286.mov

Reply:
Hey, I pushed a new commit that fixes this bug and adds cypress tests for the most common scenarios. https://github.com/elastic/kibana/pull/135210/commits/1c1c14fa05a26ffa6f59190256f30b2cc0cfb0c9
It should also fix all bugs related to loading filters when navigating between pages and refreshing the current page.

Reply:
Awesome, it works a lot better now. One last thing I noticed is this constant: `TIMELINE_FILTER_DROP_AREA` when you open a `savedQuery` in the timeline and then navigate to another app, like discover, you see this. I was unable to reproduce it in main.

<img width="1545" alt="Screen Shot 2022-07-06 at 2 26 49 PM" src="https://user-images.githubusercontent.com/17211684/177618370-6ac1c7a1-8885-49a6-a49e-ba08c7851cfb.png">

Reply:
I am resolving the discussion because this bug is reproducible in the main branch.